### PR TITLE
Added revision ontology

### PR DESCRIPTION
**Pull request by daghovland**

### CODE REVIEW

**Review by hvags — CHANGES_REQUESTED** [review]
- Needs revision number as data property; probably revNo rdfs:range xsd:string
- revisedAtDate naming may be confusing; it is the date at which the revision was issued rather than revised, also rdfs:range xsd:date rather than rdfs:range xsd:dateTime
- May have a isNewRevisionOf relation to a previous business revision (always, unless it's the first business revision issued)
- Consider moving document specific properties such as issued by, reason for issue, etc. to a "documentRevision" subclass
- does the revision object need an identifier or is that covered by the record in which it is stored?

**Comment by daghovland** [comment]
> * Needs revision number as data property; probably revNo rdfs:range xsd:string
> * revisedAtDate naming may be confusing; it is the date at which the revision was issued rather than revised, also rdfs:range xsd:date rather than rdfs:range xsd:dateTime
> * May have a isNewRevisionOf relation to a previous business revision (always, unless it's the first business revision issued)
> * Consider moving document specific properties such as issued by, reason for issue, etc. to a "documentRevision" subclass
> * does the revision object need an identifier or is that covered by the record in which it is stored?

I think this is covered in the new suggestion (current files in this PR)

**Review by hvags — COMMENTED** [review]
* prefixes
* revision:describes
* tagnumber
* containsRecords

File: schema/revision.shacl [hunk 1]
```diff
@@ -5,46 +5,46 @@
...
-        sh:message "A document edition has exactly one revision number";
+        sh:message "A document revision has exactly one revision number";
     ] ,
     [
         sh:path rev:issuedAtDate;
         sh:minCount 1;
         sh:maxCount 1;
         sh:name "RevisionDate";
-        sh:message "A documnt edition has exactly one date";
+        sh:message "A documnt revision has exactly one date";
```
Comment (markus-ap) [inline schema/revision.shacl:44]:
```suggestion
        sh:message "A document revision has exactly one date";
```
typo

File: schema/revision.shacl [hunk 2]
```diff
@@ -5,46 +5,46 @@
...
     ] ,
     [
         sh:path rev:issuedBy;
         sh:minCount 1;
         sh:name "IssuedBy";
-        sh:message "A documnt edition is issued by at least one author";
+        sh:message "A documnt revision is issued by at least one author";
```
Comment (markus-ap) [inline schema/revision.shacl:50]:
```suggestion
        sh:message "A document revision is issued by at least one author";
```
typo

**Review by markus-ap — COMMENTED** [review]
Just some suggestions for fixing typos

File: example/revisions.trig [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        prov:generatedAtTime "2022-11-03"^^xsd:date;
+                        rec:describes document:B123-EX-W-LA-00001;
+                        rec:isSubRecordOf exdata:RecordID123.
+    document:B123-EX-W-LA-00001 a <https://rdf.equinor.com/ontology/mel/EquipmentList>;
+                                rdfs:label "B123-EX-W-LA-00001"^^xsd:string;
+                                famvocab:DocumentType "LA"^^xsd:string;
+                                famvocab:Facility facility:EXMPL;
+                                famvocab:FunctionalSystem fsystem:ENG;
+                                famvocab:Location "Northsea"^^xsd:string;
+                                famvocab:Title "MASTER EQUIPMENT LIST - MEL"^^xsd:string.
```
Comment (bhtvedt) [inline example/revisions.trig:38]: The naming is a work in progress, but I think we should at least cover what's in the TIE message today:
                INST_CODE
                PROJECT_CODE
                CONTR_CODE
                DOC_NO
                DOC_TITLE
                **REV_NO**
                DOC_CATEGORY
                DISCIPLINE_CODE
                DOC_TYPE
                **REV_STATUS**
                **REV_DATE**
                **REASON_FOR_ISSUE**
                DOC_GR
                **PURPOSE_CODE**
                CONTRACT_NO
                TRANSFER_CODE

The **bold** codes belongs on the DocumentRevision object. 

PS: I do not suggest using the names above. This is just what they are known as in the TIE_MESSAGE

File: example/revisions.trig [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                        prov:generatedAtTime "2022-11-03"^^xsd:date;
+                        rec:describes document:B123-EX-W-LA-00001.F01;
+                        rec:isSubRecordOf exdata:RecordID123.
+    document:B123-EX-W-LA-00001.F01 a rev:DocumentRevision;
+                                    rdfs:label "First delivered revision of MEL"^^xsd:string;
+                                    rev:revisionNumber "F01"^^xsd:string;
+                                    prov:generatedAtTime "2022-11-03"^^xsd:date;
+                                    rev:issuedBy "Kari Nordkvinne"^^xsd:string;
+                                    rev:isNewRevisionOf document:B123-EX-W-LA-00001.A01;
+                                    famvocab:DisciplineApprovedBy "NM"^^xsd:string;
```
Comment (bhtvedt) [inline example/revisions.trig:52]: rev:IssuedBy and famvocab:DisciplineApprovedBy is not information that is currently part of the Tie Message. Include for later use?

File: example/revisions.trig [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                                    project:L.01234A.001.
+}
+
+exdata:RecordID123.0 {
+    exdata:RecordID123.0 a rec:Record;
+                        rdfs:comment "First MEL delivered on Example Facility"^^xsd:string;
+                        prov:generatedAtTime "2022-11-03"^^xsd:date;
+                        rec:describes document:B123-EX-W-LA-00001;
+                        rec:isSubRecordOf exdata:RecordID123.
+    document:B123-EX-W-LA-00001 a <https://rdf.equinor.com/ontology/mel/EquipmentList>;
```
Comment (bhtvedt) [inline example/revisions.trig:32]: Most, but not necessarily all, things in the MEL are Equipment. Could we for instance call this document:B123-EX-W-LA-00001 a <https://rdf.equinor.com/ontology/report/mel>?

File: schema/revision.shacl [hunk 6]
```diff
@@ -0,0 +1,58 @@
+@prefix sh: <http://www.w3.org/ns/shacl#>.
```
Comment (bhtvedt) [inline schema/revision.shacl:1]: Really like the Shaql rules 😄 

Add one for the additional document info, i.e. report:mel?

File: schema/revision.ttl [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+###  http://www.w3.org/2004/02/skos/core#prefLabel
+skos:prefLabel rdf:type owl:AnnotationProperty .
+
+
+#################################################################
+#    Datatypes
+#################################################################
+
+###  http://www.w3.org/2001/XMLSchema#date
+xsd:date rdf:type rdfs:Datatype .
```
Comment (bhtvedt) [inline schema/revision.ttl:29]: Do we need to type xsd - types? Remove?

File: schema/revision.ttl [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+
+#################################################################
+#    Object Properties
+#################################################################
+
+###  http://www.w3.org/ns/prov#used
+<http://www.w3.org/ns/prov#used> rdf:type owl:ObjectProperty ;
```
Comment (bhtvedt) [inline schema/revision.ttl:37]: I cannot see that this is used, remove?

File: schema/revision.ttl [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                                                         rdfs:range <https://rdf.equinor.com/ontology/record/Record> ;
+                                                         rdfs:comment "A revision contains one or more records. They represent the imutable state which the revision author(s) used for the revision process." .
+
+
+#################################################################
+#    Data properties
+#################################################################
+
+###  http://www.w3.org/ns/prov#endedAtTime
+<http://www.w3.org/ns/prov#endedAtTime> rdf:type owl:DatatypeProperty ;
```
Comment (bhtvedt) [inline schema/revision.ttl:68]: Do we use this? Remove?

**Review by bhtvedt — DISMISSED** [review]
This is great. Awesome starting point 🤘 

I approve as this will get us a big step further in establishing the pipeline.
The comments about scope and identifying records as MEL must somehow be solved, but not as part of this PR.

File: schema/revision.ttl [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+
+###  https://rdf.equinor.com/ontology/revision/isNewRevisionOf
+:isNewRevisionOf rdf:type owl:ObjectProperty ;
+                rdfs:subPropertyOf <http://www.w3.org/ns/prov#wasDerivedFrom> ;
+                rdfs:domain <https://rdf.equinor.com/ontology/revision/Revision> ;
+                rdfs:range <https://rdf.equinor.com/ontology/revision/Revision> .
+
+###  https://rdf.equinor.com/ontology/revision/describes
+:describes rdf:type owl:ObjectProperty ;
```
[the remark this answers is not in the document]
Comment (daghovland) [reply schema/revision.ttl:50]: No. rec:describes is used for record (in the provenance) while rev:describes connects a revision and a document (in the content)

File: example/revisions.trig [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+@prefix famdata: <https://rdf.equinor.com/fam/>.
+@prefix contract: <https://rdf.equinor.com/fam/Contract/>.
+@prefix document: <https://rdf.equinor.com/fam/Document/>.
+@prefix facility: <https://rdf.equinor.com/fam/Facility/>.
+@prefix project: <https://rdf.equinor.com/fam/Project/>.
+@prefix fsystem: <https://rdf.equinor.com/fam/FunctionalSystem/>.
+@prefix famvocab: <https://rdf.equinor.com/ontology/fam/v1/>.
+@prefix prov: <http://www.w3.org/ns/prov#>.
+
+exdata:RecordID123 {
```
[the remark this answers is not in the document]
Comment (daghovland) [reply example/revisions.trig:16]: Not sure what you mean by removed.

File: example/revisions.trig [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+    exdata:RecordID123 a rec:Record;
+                        rdfs:comment "Container record, defines scope for subrecords"^^xsd:string;
+                        prov:generatedAtTime "2022-11-03"^^xsd:date;
+                        rec:isInScope contract:12345678,
+                                    document:B123-EX-W-LA-00001,
+                                    facility:EXMPL,
+                                    project:L.01234A.001.
```
[the remark this answers is not in the document]
Comment (daghovland) [reply example/revisions.trig:23]: I agree but I think it belongs in a separate class and perhaps even in a different repo

File: example/revisions.trig (hunk 3 above)
Comment (daghovland) [reply example/revisions.trig:38]: Agree again, but I need to check if that should be in a separate repo

File: example/revisions.trig (hunk 4 above)
Comment (daghovland) [reply example/revisions.trig:52]: These are just meant as examples of "extra data" on a revision

File: schema/revision.ttl (hunk 8 above)
Comment (daghovland) [reply schema/revision.ttl:37]: Good point. Removed

File: schema/revision.ttl (hunk 9 above)
Comment (daghovland) [reply schema/revision.ttl:61]: Good catch. Used previously by a different revision model

**Comment by daghovland** [comment]
> * prefixes
> * revision:describes
> * tagnumber
> * containsRecords

This is all included now

File: example/revisions.trig (hunk 11 above)
[the remark this answers is not in the document]
Comment (daghovland) [reply example/revisions.trig:16]: I think this is covered

File: example/revisions.trig (hunk 3 above)
Comment (daghovland) [reply example/revisions.trig:38]: Date and Number are covered. I think status and purpose code should be subclasses or object relation. Will cover in new PR

File: schema/revision.ttl [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+                rdfs:comment "This name or number identifies the Revision in a series of Revisions." .
+
+
+###  https://rdf.equinor.com/ontology/revision/issuedBy
+:issuedBy rdf:type owl:DatatypeProperty ;
+          rdfs:subPropertyOf prov:wasAttributedTo ;
+          rdfs:domain <https://rdf.equinor.com/ontology/revision/DocumentRevision> ;
+          rdfs:range xsd:string ;
+          rdfs:comment "The author or responsible for the Revision" .
+
```
[the remark this answers is not in the document]
Comment (daghovland) [reply schema/revision.ttl:80]: I will create these in separate PR

File: schema/revision.ttl (hunk 7 above)
Comment (daghovland) [reply schema/revision.ttl:29]: This is generated from protege, but lets try to remove it.

File: schema/revision.ttl [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+<https://rdf.equinor.com/ontology/revision> rdf:type owl:Ontology ;
+                                             owl:versionIRI <https://rdf.equinor.com/ontology/revision/2023-01-10> ;
+                                             owl:imports <https://rdf.equinor.com/ontology/record-syntax/2022-12-05> .
+
+#################################################################
+#    Annotation properties
+#################################################################
+
+###  http://www.w3.org/2004/02/skos/core#prefLabel
+skos:prefLabel rdf:type owl:AnnotationProperty .
```
[the remark this answers is not in the document]
Comment (daghovland) [reply schema/revision.ttl:21]: done

File: schema/revision.shacl (hunk 6 above)
Comment (daghovland) [reply schema/revision.shacl:1]: Lets take that in a separate ontology

File: schema/revision.ttl [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix prov: <http://www.w3.org/ns/prov> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix prov: <http://www.w3.org/ns/prov#> .
+@base <https://rdf.equinor.com/ontology/revision> .
+
+<https://rdf.equinor.com/ontology/revision> rdf:type owl:Ontology ;
+                                             owl:versionIRI <https://rdf.equinor.com/ontology/revision/2023-01-10> ;
+                                             owl:imports <https://rdf.equinor.com/ontology/record-syntax/2022-12-05> .
```
[the remark this answers is not in the document]
Comment (daghovland) [reply schema/revision.ttl:14]: Ok done

File: example/revisions.trig (hunk 5 above)
Comment (daghovland) [reply example/revisions.trig:32]: Ok will do separately

**Pull request merged**